web/app/controller/Root.js

```diff
@@ -34,11 +34,17 @@ Ext.define('Traccar.controller.Root', {
     },
 
     onServerReturn: function (options, success, response) {
+        var tokenParameter;
         if (success) {
             Traccar.app.setServer(Ext.decode(response.responseText));
+            tokenParameter = window.location.search.match('token=([a-zA-Z0-9]+)');
             Ext.Ajax.request({
                 scope: this,
                 url: 'api/session',
+                method: 'GET',
+                params: tokenParameter !== null ? {
+                    token: tokenParameter[1]
+                } : {},
```

Comment:
I think you can do something like this instead:
```javascript
var token, parameters = {};
...
token = ...;
if (token) {
    parameters.token = token;
}
```

Also, I think there is a function to parse query parameters in ExtJS. Cleaner than using regex.